src/Mvc/Mvc.Core/test/Infrastructure/JsonResultExecutorTestBase.cs

```diff
@@ -68,6 +68,11 @@ public async Task ExecuteAsync_NullEncoding_DoesNotSetCharsetOnContentType()
         [Fact]
         public async Task ExecuteAsync_UsesEncodingSpecifiedInContentType()
         {
+            if (!SerializesWithContentTypeEncoding)
```

Comment:
Can you make these virtual, override them in the derived type and skip? We do this for other tests in MVC and it avoids polluting the base type with derived type specific details.